doc: Add PHP section to index (#86401)
NixOS/nixpkgs: doc/languages-frameworks/php.section.md

> # PHP

Reviewer: There is a pandoc extension, not sure if enabled by default, that allows defining ids. All headings need to have IDs specified to maintain permanent reference.

```suggestion
# PHP {#sec-language-php}
```

Author: Seems to work, added to all headings in the document :)

> ## Using PHP

Reviewer: ```suggestion
## User Guide {#ssec-php-user-guide}
```

`php.extensions.opcache` and the third-party ImageMagick extension at
`php.extensions.imagick`.

> ### Installing PHP with extensions {#sssec-php-user-guide-installing-with-extensions}

Reviewer: ```suggestion
### Installing PHP with extensions {#ssec-php-user-guide-installing-with-extensions}
```

}
```

> #### Example setup for `phpfpm` {#ssssec-php-user-guide-installing-with-extensions-phpfpm}

Reviewer: ```suggestion
#### Example setup for `phpfpm` {#sssec-php-user-guide-installing-with-extensions-phpfpm}
```

Reviewer: Or maybe just

```suggestion
#### Example setup for `phpfpm` {#ssec-php-user-guide-installing-with-extensions-phpfpm}
```

sub-sub-section is still a sub-section after a fashion.

```sh
nix-shell -p 'php.withExtensions ({ all, ... }: with all; [ imagick opcache ])'
```

### Installing PHP packages with extensions

> All interactive tools are built on our default PHP package that has a
> default set of extensions enabled. But sometimes this default set isn't
> enough and you may want to extend it.
> 
> A common case of that is the `composer` package since a project may
> depend on certain extensions and then `composer` won't work very well
> for that project unless it has those extensions loaded.
> 
> Example of building `composer` with extra extensions added:
> ```nix
> (php.withExtensions ({all, enabled }:
>   enabled ++ (with all; [ imagick redis ]))

Reviewer 2: ```suggestion
All interactive tools use the PHP package you get them from, so all
packages at `php.packages.*` use the `php` package with its default
extensions. Sometimes this default set of extensions isn't enough and
you may want to extend it. A common case of this is the `composer`
package: a project may depend on certain extensions and `composer`
won't work with that project unless those extensions are loaded.

Example of building `composer` with additional extensions:
```nix
(php.withExtensions ({ all, enabled }:
  enabled ++ (with all; [ imagick redis ]))
```

Author: Added :)
